### ISSUE TITLE
[Android] Screen Rotation

### ISSUE THREAD

**Issue by maveroke** [alexrainman/CarouselView#472]
I have view elements other than an image in the carousel view. It all works fine on iOS but when rotated to landscape on Droid and then back to portrait the carousel view doesn't redraw its child elements to the full height.

It appears SetNativeView() is getting called too early resulting in the UI being cut off after rotation.
I have a fix if you want to have a look. Ive tested it on the demo and a few other code examples people left.

Happy to be wrong as well though if I just missed something.

`
<Grid>
        <cv:CarouselViewControl Orientation="Horizontal" ItemsSource="{Binding Pages}">
            <cv:CarouselViewControl.ItemTemplate>
                <DataTemplate>
                    <StackLayout HorizontalOptions="FillAndExpand" VerticalOptions="FillAndExpand">
                        <Label Text="test"/>
                        <ListView ItemsSource="{Binding Demos}" HorizontalOptions="FillAndExpand" VerticalOptions="FillAndExpand">
                            <ListView.ItemTemplate>
                                <DataTemplate>
                                    <TextCell Text="{Binding Name}" Detail="{Binding Detail}"/>
                                </DataTemplate>
                            </ListView.ItemTemplate>
                        </ListView>
                    </StackLayout>
                </DataTemplate>
            </cv:CarouselViewControl.ItemTemplate>
        </cv:CarouselViewControl>
    </Grid>
`

`

    public class MainViewModel : INotifyPropertyChanged
    {
        public event PropertyChangedEventHandler PropertyChanged;

        public List<Demo> Demos { get; set; }
        public List<MyPage> Pages { get; set; }
        public MainViewModel()
        {
            Demos = new List<Demo>
            {
                new Demo { Name = "Apple", Detail = "Good" },
                new Demo { Name = "Banana", Detail = "Good" },
                new Demo { Name = "Carrot", Detail = "Good" },
                new Demo { Name = "Dog Food", Detail = "Bad" },
                new Demo { Name = "Eggs", Detail = "Neutral" },
                new Demo { Name = "Fries", Detail = "Good" },
                new Demo { Name = "Grass", Detail = "Bad" },
                new Demo { Name = "Hail", Detail = "Bad" },
                new Demo { Name = "Interstate", Detail = "Neutral" },
                new Demo { Name = "Jaguar", Detail = "Good" }
            };
            Pages = new List<MyPage>
            {
                new MyPage { Name = "Red", Demos = Demos },
                new MyPage { Name = "Green", Demos = Demos },
                new MyPage { Name = "Blue", Demos = Demos },
                new MyPage { Name = "???", Demos = Demos }
            };
        }

        protected virtual void OnPropertyChanged(string propertyName)
        {
            PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
        }
    }

    public class MyPage
    {
        public string Name { get; set; }

        public List<Demo> Demos { get; set; }
    }

    public class Demo
    {
        public string Name { get; set; }
        public string Detail { get; set; }
    }
`

**Comment by hectorast** [comment]
I have the same problem, I tried this solution but doesn't work.
To reproduce this issue you can modifiy this in demo project:

`<?xml version="1.0" encoding="UTF-8"?>
<ContentPage xmlns="http://xamarin.com/schemas/2014/forms"
    xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
    x:Class="Demo.MainPage"
    xmlns:cv="clr-namespace:CarouselView.FormsPlugin.Abstractions;assembly=CarouselView.FormsPlugin.Abstractions"
    xmlns:ffimageloading="clr-namespace:FFImageLoading.Forms;assembly=FFImageLoading.Forms">

    <cv:CarouselViewControl x:Name="carousel"
        ItemsSource="{Binding MyItemsSource}"
        ShowArrows="true"
        ShowIndicators="true"
        PositionSelectedCommand="{Binding MyCommand}"
        PositionSelected="Handle_PositionSelected"
        Scrolled="Handle_Scrolled"
        Orientation="Horizontal">
        <cv:CarouselViewControl.ItemTemplate>
            <DataTemplate>
                <Grid>
                    <Grid.RowDefinitions>
                        <RowDefinition Height="*"/>
                        <RowDefinition Height="80"/>
                    </Grid.RowDefinitions>
                    <Grid Grid.Row="0">
                    <ffimageloading:CachedImage DownsampleToViewSize="true" Source = "{Binding }" Aspect="AspectFill" />
                    </Grid>
                    <Grid Grid.Row="1" HeightRequest="80" BackgroundColor="White"/>
                </Grid>
            </DataTemplate>
        </cv:CarouselViewControl.ItemTemplate>

    </cv:CarouselViewControl>

</ContentPage>`

And in the ViewModel

`using System;
using System.Collections.ObjectModel;
using System.ComponentModel;
using System.Diagnostics;
using System.Windows.Input;
using FFImageLoading.Forms;
using Xamarin.Forms;

namespace Demo
{
    public class MainViewModel : INotifyPropertyChanged
    {
        public event PropertyChangedEventHandler PropertyChanged;

        public MainViewModel()
        {


            MyItemsSource = new ObservableCollection<String>()
            {
                "c1.jpg",
                "c2.jpg",
                "c3.jpg"
            };

            MyCommand = new Command(() =>
            {
                Debug.WriteLine("Position selected.");
            });
        }
        /*
        ObservableCollection<View> _myItemsSource;
        public ObservableCollection<View> MyItemsSource { 
            set {
                _myItemsSource = value;
                OnPropertyChanged("MyItemsSource");
            }
            get {
                return _myItemsSource;
            }
        }*/

        ObservableCollection<String> _myItemsSource;
        public ObservableCollection<String> MyItemsSource
        {
            set
            {
                _myItemsSource = value;
                OnPropertyChanged("MyItemsSource");
            }
            get
            {
                return _myItemsSource;
            }
        }

        public Command MyCommand { protected set; get; }

        protected virtual void OnPropertyChanged(string propertyName)
        {
            PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
        }
    }
}
`

**Comment by Kiitoksia** [comment]
I'm still getting this issue on the latest version (5.2.0).  Child elements don't re-draw themselves and occupy the same space/position as the previous rotation.

**Comment by lazarnikolov94** [comment]
I can also confirm. In my case, initially, the carousel view doesn't load the first step, but if I change the orientation it loads the list in Landscape mode. After that, if I rotate it back to Portrait mode, the list (first step) stays in Landscape mode.

**Comment by alexrainman** [comment]
@Kiitoksia no, the project is not abandoned. Next version will be out soon.

**Comment by lazarnikolov94** [comment]
@Kiitoksia thanks for the fix! It works great!
